cub/test/test_device_segmented_reduce_offset_type_fail.cu

```diff
@@ -0,0 +1,30 @@
+#include <cub/device/device_segmented_reduce.cuh>
+
+int main()
+{
+  using offset_t = float; // error
+  // using offset_t = int; // ok
+  float *d_in{}, *d_out{};
+  offset_t* d_offsets{};
+  std::size_t temp_storage_bytes{};
+  std::uint8_t* d_temp_storage{};
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::Reduce(
+    d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1, cub::Min(), 0);
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::Sum(d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1);
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::Min(d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1);
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::ArgMin(d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1);
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::Max(d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1);
+
+  // expected-error {{"Offset iterator type should be integral."}}
+  cub::DeviceSegmentedReduce::ArgMax(d_temp_storage, temp_storage_bytes, d_in, d_out, 0, d_offsets, d_offsets + 1);
```

Comment:
important: currently, this works as follows: CMake builds this targets, searches for the regex in the `{{ ... }}`, and treats the test as passed if the compilation output contains the regex. As written, all API would appear to produce an error even if only a single one of those actually produced it. I don't know a better solution than splitting this into five standalone tests.